bindings/go/constructor/construct.go

```diff
@@ -419,11 +413,39 @@ func (c *DefaultConstructor) processResource(ctx context.Context, targetRepo Tar
 	}
 
 	logger.Debug("resource processed successfully")
+
+	// update resource in graph
+	if err := c.graph.WithWriteLock(func(d *dag.DirectedAcyclicGraph[string]) error {
```

Comment:
trying to set additional metadata while constructing (doesn't work, only an experiment)